src/processor/operator/hash_join/join_hash_table.cpp

```diff
@@ -49,12 +38,15 @@ void JoinHashTable::appendVectors(const std::vector<ValueVector*>& keyVectors,
     auto numTuplesToAppend = keyState->selVector->selectedSize;
     auto appendInfos = factorizedTable->allocateFlatTupleBlocks(numTuplesToAppend);
     auto colIdx = 0u;
+    std::vector<ValueVector*> dummyUnFlatKeyVectors;
+    computeVectorHashes(keyVectors, dummyUnFlatKeyVectors);
```

Comment:
normally I would expose another `computeVectorHashes(flatKeyVectors)` in the base class. And comment we assume all keys to be flat with this API.